airbyte-connector-builder-server/connector_builder/impl/default_api.py

```diff
@@ -190,7 +198,8 @@ def _get_message_groups(self, messages: Iterator[AirbyteMessage], schema_inferre
         else:
             if not current_page_request or not current_page_response:
                 raise ValueError("Every message grouping should have at least one request and response")
-            yield StreamReadPages(request=current_page_request, response=current_page_response, records=current_records)
+            current_slice_pages.append(StreamReadPages(request=current_page_request, response=current_page_response, records=current_records))
+            yield StreamReadSlices(pages=current_slice_pages)
```

Comment:
should we also set the slice_descriptor and the state now that we're partitioning by slices?

@lmossman there aren't any short-term plans to use these fields, right?

Reply:
We already have a dropdown implemented for selecting the slice implemented which will be rendered in the UI if there is more than one slice being returned, so you should be able to see what that looks like on this PR branch. That dropdown uses the slice descriptor to decide what values to show in the dropdown.

But I don't believe we currently have any plans for `state`

Reply:
I am still unclear as to how we should populate the `slice_descriptor` but I think this can be part of another issue if/when needed

Reply:
Agreed, it doesn't look too bad to me

Reply:
A couple ideas for populating the slice_descriptor:
- If using a datetime stream slicer, it could be set to the starting datetime of the current slice, e.g. `1 Jan 2023`
- If using a list stream slicer, it could be set to the current list item, e.g. `comments`
- If using a cartesian product of datetime and list, it could be a concatenation of both, e.g. `1 Jan 2023 | comments`

Not as sure about substream, maybe it could be set to the parent slice key or something like that?